src/lucky/params.cr

```diff
@@ -145,6 +145,37 @@ class Lucky::Params
     route_params[key.to_s]? || body_param(key.to_s) || query_params[key.to_s]?
   end
 
+  # Retrieve values for a given key, returns an empty array if no values found
+  #
+  # Checks in places that could provide multiple values and returns first with values:
+  # - JSON body
+  # - multipart params
+  # - form encoded params
+  # - query params
+  #
+  # Does not check for path variables.
+  #
+  # For all params locations it appends square brackets
+  # so searching for "emails" in query params will look for values with a key of "emails[]"
+  #
+  # ```crystal
+  # params.get_all(:names)    # ["Paul", "Johnny"] : Array(String)
+  # params.get_all("missing") # [] of String : Array(String)
+  # ```
+  def get_all(key : String | Symbol) : Array(String)
```

Comment:
I keep wondering if maybe this should raise an error instead of return an empty array like how `get` does. Then have a `get_all?` that returns nil if missing. 🤔

Reply:
I wasn't sure about that. I ended up copying the way `HTTP::Params#fetch_all` works but I'd be open to changing it. I think Avram would need to do an empty check with the current change or else it would always find a value.

What's the call?

Reply:
I like that escape hatch with the `fetch_all` which could be the way out. Then I think with this, we should match up with what `get()` does now where if it's missing, `raise Lucky::MissingParamError.new(key.to_s)`, and then add a `get_all?()` that returns `Array(String) | Nil`.